src/Query/Providers/QueryProvider.cs

```diff
@@ -286,11 +286,11 @@ public async Task ExecuteInTransactionAsync(Func<Task> action)
                 try
                 {
                     await action();
-                    transaction.Commit();
+                    await transaction.CommitAsync();
```

Comment:
Neviem či to chceme/potrebujeme a či to robiť v rámci tohto, ale možno by bolo fajn posielať token zvonku do takejto metódy, tzn. pridať novú `public async Task ExecuteInTransactionAsync(Func<CancellationToken, Task> action, CancellationToken token = default)`. Token by sa poslal do commit/rollback, aj do tej akcie.